app/controllers/api/allocation_api_controller.rb

```diff
@@ -6,6 +6,9 @@ def show
       render_404('Not ready for allocation') && return if allocation.nil?
       render_404('Not allocated') && return if allocation.primary_pom_nomis_id.nil?
 
+      offender = OffenderService.get_offender(offender_number)
+      render_404('Not sentenced') && return if not offender.sentenced?
```

Comment:
This introduces a new error response body – `"Not sentenced"`.

Since this API is consumed by DPS, I wonder how risky it is to do this.

`"Not allocated"` could also be valid to use here, which means we don't introduce a new response body.

But: again, it's worth checking with @starswan because he'll understand this more than I do!